packages/clay-layout/src/Content.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+interface IRowProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Element or component to render for container
+	 */
+	containerElement?: React.JSXElementConstructor<{
+		className: string;
+		[key: string]: any;
+	}>;
+
+	/**
+	 * Provides the benefit of aligning content via flexbox without losing the behavior
+	 * of floated elements at the expense of extra markup.
+	 */
+	float?:
+		| boolean
+		| 'sm-down'
+		| 'md-down'
+		| 'end'
+		| 'end-sm-down'
+		| 'end-md-down';
+
+	/*
+	 * Give negative margins on the top, right, bottom, and left to offset the padding
+	 */
+	noGutters?: 'x' | 'y' | true;
+
+	/*
+	 * Gives padding to all autofit-cols that are direct children of autofit-row.
+	 */
+	padded?: boolean;
+
+	/**
+	 * Adds class for aligning items within the row.
+	 */
+	verticalAlign?: 'center' | 'end';
+}
+
+const ContentRow: React.FunctionComponent<IRowProps> = ({
+	children,
+	className,
+	containerElement: ContainerElement = 'div',
+	float,
+	noGutters,
+	padded,
+	verticalAlign,
+	...otherProps
+}) => {
+	return (
+		<ContainerElement
+			{...otherProps}
+			className={classNames(className, 'autofit-row', {
+				'autofit-float': float === true,
+				[`autofit-float-${float}`]: typeof float === 'string',
+				'autofit-padded': padded,
+				'autofit-padded-no-gutters': noGutters === true,
+				[`autofit-padded-no-gutters-${noGutters}`]:
```

Comment:
Hey @pat270, one doubt here... is it `autofit-padded` always needed with `autofit-padded-no-gutters`? Is there any logic here that needs to be coded?

Meaning, is just `autofit-padded-no-gutters` correct? or setting `-no-gutters` implies `padded`?

@bryceosterhaus, just in case those 2 props are in fact related...

Reply:
@jbalsas @bryceosterhaus 
No `autofit-padded` isn't always needed with `autofit-padded-no-gutters`. I added the same padding to both because it gets confusing. You can use `autofit-padded autofit-padded-no-gutters` or just `autofit-padded-no-gutters`.

I think the most common use case would be `autofit-row autofit-padded-no-gutters-x` which uses negative margins to align the left and right side of `autofit-row` (similar to Bootstrap's `row`), but provides a little white space top and bottom. 

---

`autofit-padded` adds `padding: 0.25rem 0.5rem` on `autofit-col`

---

`autofit-padded-no-gutters-x` adds `padding: 0.25rem 0.5rem` on `autofit-col` and `margin-left: -0.5rem; margin-right: -0.5rem` on `autofit-row`

---

`autofit-padded-no-gutters` adds `padding: 0.25rem 0.5rem` on `autofit-col` and `margin: -0.25rem -0.5rem` on `autofit-row`